### DISCUSSION TITLE
Could not find an attached device

### DISCUSSION THREAD

**Discussion by chadoulis** [STELLA-Landsat/STELLA#25] [Build Notes: How do I make a STELLA?]
On step 9 of programming instructions, I click on the Serial button in Mu, to look at the output of the STELLA instrument. Initially, it seems like the device has already been detected, at least according to the corresponding indication at the right corner of the Mu window:

![Screenshot 2023-06-12 033123---](https://github.com/STELLA-Landsat/STELLA/assets/25674261/38ff1fa2-bba8-49c8-9ef3-9518d74d00f4)

Also something like a command line section has opened up at the bottom of the Mu window:

![Screenshot 2023-06-12 0331230000000](https://github.com/STELLA-Landsat/STELLA/assets/25674261/5f018c6b-213b-40e5-8e5b-d6b22573e65d)

Usually this command line remains empty. But once or twice the following appeared there: 

```
memory used for library imports, B : 80032 which is 56.3% of total
UID: 429
```

However after a while the error message `Could not find an attached device` appears:

![Screenshot 2023-06-12 033159](https://github.com/STELLA-Landsat/STELLA/assets/25674261/0e49aece-6b6d-4eba-a1b7-887edf2728a5)

Any ideas?

**Comment by paulmirel** [comment]
I can't recreate that error on my Windows system. 

What version of Mu are you running? I've got 1.0.2, which is shown on the upper left corner of the main Mu window. It looks different from yours: the legend on the serial dialogue window is "Adafruit CircuitPython REPL"

What version of the STELLA-2.0 software are you running? Open the code.py file, and the first line or two should say the version. I've gotten the STELLA to talk to Mu with both v2.0 and v2.1.0.

**Comment by paulmirel** [comment]
I upgraded to Mu 1.2.0, and it continues to work. The STELLA mirror data show up in the Serial Dialogue panel.

**Reply by chadoulis** [reply]
- Mu 1.2.0
 - STELLA-2 instrument code v2.0
 
 Also, something that may be of importance. I 've tried hard reset a few times. It goes like this: 
 
 1) I press the reset button twice.
 2) The Featherboot folder containing the `.uf2` file appears. 
 3) I drag and drop the .uf2 file I downloaded from the official website according to the model of my microprocessor which is a slight variation of the suggested one (`adafruit-circuitpython-feather_nrf52840_express-en_US-8.1.0`). 
 4) The copying process gets interrupted with a warning about the drive not being found.
 5) Sometimes the `CIRCUITPY `folder appears, but most of the times it doesn't. So I keep plugging and unplugging until it does. And then I copy the `code.py` file and the `lib `folder in the `CIRCUITPY `drive.
 6) Then I open the Mu Editor and try to follow the next steps...

**Comment by paulmirel** [comment]
I don't know offhand how robust the software is against finding a microcontroller that doesn't have the sensors on board. 

The BOOT drive disappears when the processor loads CircuitPython. That error, in your line 4, is normal. 

5. Something is wrong with the microcontroller. If it's not connecting reliably, it's going to be very hard to get it to do anything else. If you are using an incorrect CircuitPython UF2 file, that could cause those sorts of problems. Try copying the code.py file from the blink_indicator folder in test_codes. If the indicator (the red LED) doesn't blink, then you almost certainly have the wrong CircuitPython UF2 file. If it does blink, then the microcontroller is OK, and the connection problem may be something going on with your computer, and not the microcontroller.

**Reply by Landsat-Mike** [reply]
I also just made sure the STELLA 2 software is the latest version we have. Check and see if the uf2 is the most current. [https://landsat.gsfc.nasa.gov/wp-content/uploads/2023/06/S2-software.zip](https://landsat.gsfc.nasa.gov/wp-content/uploads/2023/06/S2-software.zip)

**Reply by chadoulis** [reply]
I am using the latest stable release of CircuitPython (CircuitPython 8.1.0) that, according to the official website, will work with my Feather nRF52840 Express and can be found at: [https://circuitpython.org/board/feather_nrf52840_express/](https://circuitpython.org/board/feather_nrf52840_express/). I also tried using the more recent beta release of CircuitPython (CircuitPython 8.2.0-beta.1). 

Following your suggestion, I copied the `code.py` file from the `blink_indicator` folder in `test_codes`. The red LED on the front of the breadboard started blinking, as shown in the image below. The secondary led did not.

![viber_image_2023-06-12_22-14-43-668](https://github.com/STELLA-Landsat/STELLA/assets/25674261/5d4af5d7-6984-4b6c-9768-39795c0dc978)

**Comment by paulmirel** [comment]
Photo is helpful. Multiple solder joints are showing signs of incomplete connection. For example, processor pin at C55 looks not connected, which would stop the secondary LED from blinking. C54 also looks suspect, as does C52, C47, A45, A46, and so on. Importantly, H54 looks suspect, which could lead to intermittent on and off of the STELLA, as that pin controls whether the 3.3V source is on or off (controlled by the power switch at A43). That could be the cause of it sometimes connecting and sometimes not, as it is sometimes on and sometimes off. 

The characteristic problems I am seeing are insufficient solder and insufficient wetting. All the solder joints should look like smooth volcanoes, with no holes showing. Try resoldering each joint, leaving the iron on the joint for a slow count of three, to make sure it melts thoroughly. 
https://learn.adafruit.com/adafruit-guide-excellent-soldering/common-problems

**Reply by chadoulis** [reply]
You were right. Insufficient solder and insufficient wetting were causing all this. I resoldered each joint, and the issue was resolved.

**Comment by paulmirel** [comment]
Excellent. Thanks for letting me know it turned out well.
